components/NavigationDrawer/examples/BottomDrawerExample.swift

```diff
@@ -0,0 +1,353 @@
+ See the License for the specific language governing permissions and
+ limitations under the License.
+ */
+
+import UIKit
+
+class BottomDrawerInfiniteScrollingExample: UIViewController {
+
+  let headerViewController = DrawerHeaderViewController()
+  let contentViewController = DrawerContentTableViewController()
+
+  override func viewDidLoad() {
+    view.backgroundColor = .white
+  }
+
+  override func viewDidAppear(_ animated: Bool) {
+    showWrappedViewController();
+  }
+
+  func showWrappedViewController() {
+    let bottomDrawerViewController = MDCBottomDrawerViewController()
+    bottomDrawerViewController.mainContentViewController = contentViewController
+    bottomDrawerViewController.headerViewController = headerViewController
+    bottomDrawerViewController.trackingScrollView = contentViewController.tableView
+    present(bottomDrawerViewController, animated: true, completion: nil)
+  }
+}
+
+class BottomDrawerWithScrollableContentExample: UIViewController {
+
+  let headerViewController = DrawerHeaderViewController()
+  let contentViewController = DrawerContentWithScrollViewController()
+
+  override func viewDidLoad() {
+    view.backgroundColor = .white
+  }
+
+  override func viewDidAppear(_ animated: Bool) {
+    showWrappedViewController();
+  }
+
+  func showWrappedViewController() {
+    let bottomDrawerViewController = MDCBottomDrawerViewController()
+    bottomDrawerViewController.mainContentViewController = contentViewController
+    bottomDrawerViewController.headerViewController = headerViewController
+    bottomDrawerViewController.trackingScrollView = contentViewController.collectionView
+    present(bottomDrawerViewController, animated: true, completion: nil)
+  }
+}
+
+class BottomDrawerWithHeaderExample: UIViewController {
+
+  let headerViewController = DrawerHeaderViewController()
+  let contentViewController = DrawerContentViewController()
+
+  override func viewDidLoad() {
+    view.backgroundColor = .white
+  }
+
+  override func viewDidAppear(_ animated: Bool) {
+    showWrappedViewController()
+  }
+
+  func showWrappedViewController() {
+    let bottomDrawerViewController = MDCBottomDrawerViewController()
+    bottomDrawerViewController.mainContentViewController = contentViewController
+    bottomDrawerViewController.headerViewController = headerViewController
+    present(bottomDrawerViewController, animated: true, completion: nil)
+  }
+
+}
+
+class BottomDrawerNoHeaderExample: UIViewController {
+
+  let contentViewController = DrawerContentViewController()
+  let bottomDrawerTransitionController = MDCBottomDrawerTransitionController()
+
+  override func viewDidLoad() {
+    view.backgroundColor = .white
+  }
+
+  override func viewDidAppear(_ animated: Bool) {
+    showContentViewControllerOnly()
+  }
+
+  func showContentViewControllerOnly() {
+    contentViewController.transitioningDelegate = bottomDrawerTransitionController
+    contentViewController.modalPresentationStyle = .custom
```

Comment:
Why is this `.custom`? It seems the others don't need to override the `modalPresentationStyle` value.

Reply:
This is `.custom` because in the other methods of presentation we use `MDCBottomDrawerViewController` which encapsulates the setting of `.custom`. In this case we solely rely on the transition controller presenting a random view controller to be displayed inside a drawer seamlessly using our custom presentation controller